### DISCUSSION TITLE
how do I put an alias in addColumn

### DISCUSSION THREAD

**Discussion by eduardohr-muniz** [simolus3/drift#2755] [Q&A]
![image](https://github.com/simolus3/drift/assets/90477293/fc72eeb5-dcde-45e5-933a-7c6f24fe5ad2)
how do I put an alias in addColumn

**Comment by simolus3** [comment]
Why do you want to alias columns? When adding an expression with `addColumns`, drift will pick a short alias automatically and remember it when using `result.read`.

**Comment by darkomenx** [comment]
In my context, the aggregate SUM() function into a column of name "amount" generate a "c0" alias name...
I don't understand logic under the hood for generate this name and this is strange to work of our result object that contain "c0" column otherwise like "amount" or "sum" or "amountSum" fore example.
For the reading and comprehensive code is not so pretty cool things for work.

Is there a way to alias our column or not ? (if no way no debate and this is not a criticism)
I think that the part of this link (https://drift.simonbinder.eu/docs/dart-api/select/) that dealing with this concept not corresponding our requirements of @eduardohr-muniz and me.

Thank you for your lighting @simolus3

**Comment by darkomenx** [comment]
@eduardohr-muniz perhaps I understand @simolus3 because I just discover things that can help you.

In the past, when I want to access result of query I use this :  
`row.rawData.data['categories.label']`
And with this method to access data the column name aggregate is too strange.

BUT we can use this : 
` // final amountOfThing = thing.amount.sum() `
`// ..addColumns([amountOfThing , thing.category])`
`row.read(amountOfThing) // whereas row.rawData.data['c0'] or row.rawData.data['0']`

In this context I understand @simolus3 when he say that generate name is not important because we can access data with the same name that I defined into our query construct 👍 

I'm right ?

**Reply by simolus3** [reply]
Yes exactly, the recommended approach to read expressions added to a select statement is `row.read`. Hiding the actual column name allows drift to generate succinct column names internally.

If you have a convincing use case for accessing `row.rawData` directly, I can look into ways to provide custom (or at least stable) column names. But even when that's necessary I probably prefer an API like

```dart
 row.rawData.data[row.nameFor(categories.label)]
```

> Is there a way to alias our column or not ?

At the moment, there isn't.

**Reply by darkomenx** [reply]
No I don't think so. I nothing convincing use case :) 

I began (and perhaps wrong way) to use this for transform result of query Drift to a Model class (linked with entity drift).
But since now I never had issue with this way of writing because without aggregate function into query, all the name of field have the name of the column (then name of entity field of course).

Example : 

      final query = select(items).join([
          leftOuterJoin(db.categories, db.categories.id.equalsExp(items.category), useColumns: true),
          leftOuterJoin(db.natures, db.natures.id.equalsExp(items.nature), useColumns: true),
      ]);

      var result = await query.get();

      return result.map((row) {

        return ItemModel
        (
            id:              row.rawData.data['items.id'],
            date:            DateTime.parse(row.rawData.data['items.date']),
            label:           row.rawData.data['items.label'],
            amount:          row.rawData.data['items.amount'],
            nature:          row.rawData.data['operations.nature'].toString(),
            categoryID:      row.rawData.data['categories.id'],
            categoryLabel:   row.rawData.data['categories.label'],
            createdAt:       DateTime.parse(row.rawData.data['items.created_at']),
            updatedAt:       DateTime.parse(row.rawData.data['items.updated_at']),
        );

      }).toList();

In this case, the both ways is similar for me because code is reading and comprehensive easily (this is an important argument)

BUT

Is there more optimized to use : `row.read() ` ?

**Reply by simolus3** [reply]
> In this case, the both ways is similar for me because code is reading and comprehensive easily (this is an important argument)

I agree that readability is important, but is `DateTime.parse(row.rawData.data['items.date'])` really more readable than `row.read(items.date)`? By using the column with `read()`, you'll also get a compile-time error if you ever decide to remove a column. So it's safer than the raw field access.

> Is there more optimized to use : `row.read()` ?

It shouldn't make a noticeable difference. Drift will have to lookup the underlying column name, but this gets cached so there is no big impact (even though `row.rawData.data[]` is doing less work).

**Reply by darkomenx** [reply]
> you'll also get a compile-time error if you ever decide to remove a column. So it's safer than the raw field access

This is a powerful argument because I lost many times with this because I have a model that lot of change due to begin dev phase.
I begin to use try/catch in my code for getting compile-time error but you alright.

Now I will use `row.read()`

Thank you @simolus3 for your support and I hope that can be helping @eduardohr-muniz too...
